tests/test_orquestra_device.py

```diff
@@ -50,24 +54,36 @@ def circuit():
         with pytest.raises(NotImplementedError):
             circuit()
 
-    def test_qasm_simulator_analytic_warning(self):
-        """Test that a warning is raised when using the QeQiskitDevice with the
-        qasm_simulator backend in analytic mode and that we'll switch to
+    @pytest.mark.parametrize("dev,backend", non_analytic_backends)
+    def test_analytic_warning(self, dev, backend):
+        """Test that a warning is raised when setting devices with
+        sampling-only backends to analytic mode and that we'll switch to
         sampling mode."""
         with pytest.warns(
             Warning,
             match="cannot be used in analytic",
         ):
-            dev = qml.device("orquestra.qiskit", backend="qasm_simulator", wires=2, analytic=True)
+            dev = qml.device(dev, backend=backend, wires=2, analytic=True)
+
+        assert not dev.analytic
+
+    def test_statevector_simulator_single_shot_warning(self):
+        """Test that a warning is raised when using the QeQiskitDevice with the
+        statevector_simulator backend in sampling mode that the shots=1."""
```

Comment:
This needs a rewrite (I think you mixed two sentences into one)!

Reply:
Good catch, thanks! Updated.